hat/assets/js/apps/Iaso/components/forms/AsyncSelect.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { FunctionComponent, useEffect, useMemo, useState } from 'react';
+import { debounce } from '@mui/material/utils';
+import { Autocomplete } from '@material-ui/lab';
+import { IntlMessage, useSafeIntl } from 'bluesquare-components';
+import { Box, Chip, TextField } from '@material-ui/core';
+import { isArray } from 'lodash';
+
+type Props = {
+    value: any;
+    label: IntlMessage;
+    loading?: boolean;
+    loadingText?: IntlMessage;
+    keyValue: string;
+    // eslint-disable-next-line no-unused-vars
+    onChange: (keyValue, newValue: any | null) => void;
+    errors?: string[];
+    required?: boolean;
+    clearable?: boolean;
+    debounceTime?: number; // debounce time in ms
+    disabled?: boolean;
+    multi?: boolean;
+    helperText?: string;
+    minCharBeforeQuery?: number;
+    // eslint-disable-next-line no-unused-vars
+    fetchOptions: (input: string) => Promise<any[]>;
+};
+
+export const AsyncSelect: FunctionComponent<Props> = ({
+    value,
+    onChange,
+    keyValue,
+    label,
+    loading = false,
+    loadingText = undefined,
+    required = false,
+    clearable = false,
+    debounceTime = 0,
+    disabled = false,
+    multi = false,
+    helperText = undefined,
+    minCharBeforeQuery = 1,
+    fetchOptions,
+}) => {
+    const { formatMessage } = useSafeIntl();
+    const [inputValue, setInputValue] = useState('');
+    const [isLoading, setLoading] = useState(loading);
```

Comment:
you can type the `useSate` here

Reply:
Isn't it automatically typed by the `loading` value I passed to it?